tests/operator_test.go

```diff
@@ -97,6 +99,114 @@ var _ = Describe("Operator tests", func() {
 		Expect(conditionMap[conditions.ConditionProgressing]).To(Equal(corev1.ConditionFalse))
 		Expect(conditionMap[conditions.ConditionDegraded]).To(Equal(corev1.ConditionFalse))
 	})
+
+	It("should deploy components that tolerate CriticalAddonsOnly taint", func() {
+		var err error
+		cr, err := f.CdiClient.CdiV1beta1().CDIs().Get(context.TODO(), "cdi", metav1.GetOptions{})
+		Expect(err).ToNot(HaveOccurred())
+
+		criticalAddonsToleration := corev1.Toleration{
+			Key:      "CriticalAddonsOnly",
+			Operator: corev1.TolerationOpExists,
+		}
+
+		if !tolerationExists(cr.Spec.Infra.Tolerations, criticalAddonsToleration) {
+			Skip("Unexpected CDI CR (not from cdi-cr.yaml), doesn't tolerant CriticalAddonsOnly")
+		}
+
+		cdiPods, err := f.K8sClient.CoreV1().Pods(f.CdiInstallNs).List(context.TODO(), metav1.ListOptions{})
+		Expect(err).ToNot(HaveOccurred(), "failed listing cdi pods")
+		Expect(len(cdiPods.Items)).To(BeNumerically(">", 0), "no cdi pods found")
+
+		labelSelector := metav1.LabelSelector{MatchLabels: map[string]string{"cdi.kubevirt.io/testing": ""}}
+		cdiTestPods, err := f.K8sClient.CoreV1().Pods(f.CdiInstallNs).List(context.TODO(), metav1.ListOptions{
+			LabelSelector: labels.Set(labelSelector.MatchLabels).String(),
+		})
+		Expect(err).ToNot(HaveOccurred(), "failed listing cdi testing pods")
+		Expect(len(cdiPods.Items)).To(BeNumerically(">", 0), "no cdi testing pods found")
+
+		By("adding taints to all nodes")
+		criticalPodTaint := corev1.Taint{
+			Key:    "CriticalAddonsOnly",
+			Value:  "",
+			Effect: corev1.TaintEffectNoExecute,
+		}
+
+		nodes, err := f.K8sClient.CoreV1().Nodes().List(context.TODO(), metav1.ListOptions{})
+		Expect(nodes.Items).ToNot(BeEmpty(), "There should be some compute node")
+		Expect(err).ToNot(HaveOccurred())
+
+		defer func() {
```

Comment:
So this is somewhat nit picky. As the defer will do the same thing. But this will be the only test that cleans up this way, where all the other tests use the AfterEach mechanism provided by Ginkgo. Maybe move this into its own describe or context, and provide the contents of the defer in an AfterEach for consistency with the other tests.